recipes/devel/ocaml.yaml

```diff
@@ -0,0 +1,44 @@
+inherit: [autotools, autoconf]
+
+metaEnvironment:
+    PKG_VERSION: "5.3.0"
+
+checkoutSCM:
+    scm: url
+    url: https://github.com/ocaml/ocaml/archive/refs/tags/${PKG_VERSION}.tar.gz
+    digestSHA256: eb9eab2f21758d3cfb1e78c7f83f0b4dd6302824316aba4abee047a5a4f85029
+    stripComponents: 1
+
+buildTools: [m4, autotools]
+buildVars: [STRIP]
+buildScript: |
+    rsync -a --delete $1/ .
+    mkdir -p .bob
+    pushd .bob
+    ln -s $(which ${STRIP}) strip
+    popd
+    export PATH=${PATH}:$(pwd)/.bob
+
+    ./configure \
```

Comment:
Is there a particular reason why you do not use `autotoolsBuild`? This applies to basically all packages.

Reply:
Yes - because `autotoolsBuild` is not working. Using `autotoolsBuild` it fails with:

 `make: *** No targets specified and no makefile found.  Stop.

The reason is, that `configure` did not produce the makefiles in the current working directory, it's always in the folder of the configure script. - but `autotoolsBuild` uses a build-folder where it runs `configure` and assumes the makefiles to be generated.

To use the code from the `autotools`-class I think we can either
- move the  `configure` part of the `autotools` - class into a separate class and reuse it from there
- add a argument to `autotoolsBuild` skipping the [pushd build](https://github.com/BobBuildTool/basement/blob/master/classes/autotools.yaml#L60)

Do you see another option? Which one do you prefer?

Reply:
Oh my gosh! It's really that borken. :facepalm:  In this case I would rather not touch the `autotools` class. But a small comment with exactly this explanation would be appreciated. Otherwise we'll start wondering again a couple of years down the road why the `autotools` class isn't use...

Reply:
I found a solution to use `autotoolsBuild` for most of the packages. It's not pretty but works...